src/QsCompiler/Core/Transformations/TypeTransformation.fs

```diff
@@ -17,22 +17,46 @@ open Microsoft.Quantum.QsCompiler.Transformations.Core.Utils
 type private ExpressionType = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>
 
 
-type TypeTransformationBase(options: TransformationOptions) =
+type TypeTransformationBase internal (options: TransformationOptions, _internal_) =
+
+    let missingTransformation name _ =
+        new InvalidOperationException(sprintf "No %s transformation has been specified." name) |> raise
+
     let Node = if options.Rebuild then Fold else Walk
 
-    new() = TypeTransformationBase TransformationOptions.Default
+    member val internal CommonTransformationItemsHandle = missingTransformation "common items" with get, set
+
+    member this.Common : CommonTransformationNodes = this.CommonTransformationItemsHandle()
+
+    internal new(getCommonItems: unit -> CommonTransformationNodes, options: TransformationOptions) as this =
+        new TypeTransformationBase(options, "_internal_")
+        then this.CommonTransformationItemsHandle <- getCommonItems
 
+    new(options: TransformationOptions) as this =
+        new TypeTransformationBase(options, "_internal_")
+        then
+            let commonItems = new CommonTransformationNodes()
+            this.CommonTransformationItemsHandle <- fun _ -> commonItems
+
+    new() = TypeTransformationBase TransformationOptions.Default
 
     // supplementary type information
 
     // TODO: RELEASE 2021-10: Remove obsolete method.
-    [<Obsolete "Use OnRangeInformation(TypeRange) instead.">]
+    [<Obsolete "Use SyntaxTreeTransformation.OnTypeRange(TypeRange) instead.">]
```

Comment:
Removing these is going to be a more involved change; the reason is that when we remove these, we also need to remove the corresponding (deprecated) handle in the AST node. That would make that record have the same items as another record type, which would break a lot of the F# code. We should properly mitigate that so that we indeed don't break anything when removing these handles. This needs to be a separate PR.